### DISCUSSION TITLE
Invertd vs. BatchInverseTransform

### DISCUSSION THREAD

**Discussion by dianemarquette** [Project-MONAI/MONAI#2332] [Q&A] [answered]
Hi! 

I'm really glad that the MONAI team developed tools to invert transforms applied to the data. It's really useful for predictions! However, I'm a bit confused about how `Invertd` differs from `BatchInverseTransform`. 

I've looked at the [unet_inference_dict.py](https://github.com/Project-MONAI/tutorials/blob/63853236d402cb84743c2d5547a66689adcc83c8/3d_segmentation/torch/unet_inference_dict.py) and the [inverse_transforms_and_test_time_augmentations.ipynb](https://github.com/Project-MONAI/tutorials/blob/63853236d402cb84743c2d5547a66689adcc83c8/modules/inverse_transforms_and_test_time_augmentations.ipynb) tutorials. But, I still can't spot the real difference between both transforms.

Thanks in advance for the clarification :)

**Comment by Nic-Ma** [comment, marked as answer]
Hi @dianemarquette ,

Thanks for your interest here.
1. Currently, `Invertd` is based on `BatchInverseTransform` and provides more features: to_tensor, device, post_func, etc. And can execute for several keys together.
2. `Invertd` is a post transform, which can be composed with other transforms.
2. In next version, we will change Invertd to execute on `channel-first` Tensor data instead of a `batch-first` Tensor.

Thanks.

**Reply by dianemarquette** [reply]
Ok, thanks for the detailed explanation :)
